host/tests/dsClasses/dsDisplay.py

```diff
@@ -0,0 +1,293 @@
+from raft.framework.plugins.ut_raft.utSuiteNavigator import UTSuiteNavigatorClass
+from raft.framework.plugins.ut_raft.interactiveShell import InteractiveShell
+
+
+class dsVideoPortType(Enum):
+    dsVIDEOPORT_TYPE_RF     = 0
+    dsVIDEOPORT_TYPE_BB =auto()
+    dsVIDEOPORT_TYPE_SVIDEO =auto()
+    dsVIDEOPORT_TYPE_1394 =auto()
+    dsVIDEOPORT_TYPE_DVI =auto()
+    dsVIDEOPORT_TYPE_COMPONENT =auto()
+    dsVIDEOPORT_TYPE_HDMI =auto()
+    dsVIDEOPORT_TYPE_HDMI_INPUT =auto()
+    dsVIDEOPORT_TYPE_INTERNAL =auto()
+
+class dsDisplayEvent(Enum):
+    dsDISPLAY_EVENT_CONNECTED = 0
+    dsDISPLAY_EVENT_DISCONNECTED = auto()
+    dsDISPLAY_RXSENSE_ON = auto()
+    dsDISPLAY_RXSENSE_OFF = auto()
+    dsDISPLAY_HDCPPROTOCOL_CHANGE = auto()
+    dsDISPLAY_EVENT_MAX = auto()
+
+class dsVideoAspectRatio(Enum):
+    dsVIDEO_ASPECT_RATIO_4x3 = auto()
+    dsVIDEO_ASPECT_RATIO_16x9 = auto()
+    dsVIDEO_ASPECT_RATIO_MAX = auto()
+
+
+class dsDisplayClass():
+
+    """
+    Device Settings Display Class
+    This module provides common extensions for device Settings Display Module.
+    """
+    moduleName = "dsDisplay"
+    menuConfig = dir_path+ "/dsDisplay_test_suite.yml"
+    testSuite = "L3 dsDisplay"
+    validationConfig = dir_path + "/../L3_TestCases/dsDisplay/dsDisplay_test_validation.yml"
+
+    def __init__(self, deviceProfilePath :str, session=None, targetWorkspace="/tmp" ):
+        """
+        Initializes the dsDisplay class function.
+        """
+
+        # Load configurations for device profile and menu
+        self.deviceProfile = ConfigRead( deviceProfilePath , self.moduleName)
+        self.validationProfile = ConfigRead(self.validationConfig,self.moduleName)
+        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session)
+        self.testSession   = session
+
+         # Start the user interface menu
+        self.utMenu.start()
+
+    def searchPattern(self, haystack, pattern):
+        """
+        Searches for the first occurrence of a specified pattern in the provided string.
+
+        Args:
+            haystack (str): The string to be searched.
+            pattern (str): The regular expression pattern to search for.
+
+        Returns:
+            str: The first capturing group of the match if found; otherwise, None.
+
+        Notes:
+            - The pattern should contain at least one capturing group (parentheses).
+            - If no match is found, None is returned.
+        """
+        match = re.search(pattern, haystack)
+        if match:
+            return match.group(1)
+        return None
+
+    def initialise(self):
+        """
+        Initializes the device settings Display module.
+
+        Args:
+            None
+
+        Returns:
+                None
+        """
+        result = self.utMenu.select( self.testSuite, "Initialize dsDisplay")
+
+    def getDisplayHandle(self, video_port: str, port_index: int=0):
```

Comment:
getDisplayHandle name bit confusing, we are not getting/returning any handle here. selectDisplayPort appropriate.